lib/reject.js

```diff
@@ -3,7 +3,9 @@ import wrap from './wrap'
 
 function reject(predicate, collection) {
   const result = _reject(collection, predicate)
-  const equal = collection.length === result.length
+  const equal = Array.isArray(collection)
+    ? collection.length === result.length
+    : Object.keys(collection).length === result.length
```

Comment:
Can you test this path as well? It looks like you may need an object.keys around result too.

Reply:
ah AH! But the new test added in `reject-specs` test it. The return value of `_.reject` is always an array.

Reply:
I think the lodash's rationale is that they say "Go on, give us a "collection", and we'll deal with it. If the collection is an array, congrat, you're making sense. If the collection is an object, okay then, we'll use the values".  Same logic that goes for `_.map` and friends.

This being said, if you prefer not to have this, it's all fine. :-)